sklearn/base.py

```diff
@@ -789,8 +789,10 @@ def fit_predict(self, X, y=None):
 
         Parameters
         ----------
-        X : {array-like, sparse matrix, dataframe} of shape \
-            (n_samples, n_features)
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            The input samples. Use ``dtype=np.float32`` for maximum
+            efficiency. Sparse matrices are also supported, use sparse
+            ``csc_matrix`` for maximum efficiency.
```

Comment:
This is a tricky one, some child classes support sparse, i.e. `IsolationForest`, but others such as `EllipticEnvelope` does not.

For now, to pass validation, we can just add "The input samples." and everything else the same?

Reply:
Thank you for the review! I should have checked the child classes. 

The comment is updated as suggested :)